### PR TITLE
[#5977] Update projects to .Net 6

### PR DESCRIPTION
**Pull request by ceciliaavila**
Addresses # 5977
#minor

## Description
This PR updates the test projects, the test bots, and the CI pipelines to target .NET6.
All these changes need to be in one PR in order for the build checks to pass. 
We divided the changes into three commits to ease the review:

- Updated test projects ([930e801](https://github.com/southworks/botbuilder-dotnet/pull/458/commits/930e80153977b1204b5ac2cde55fb624b3f7f174))
- Updated test bots ([53265ad](https://github.com/southworks/botbuilder-dotnet/pull/458/commits/53265ad41a26b6f2f28e3bbb723ddc1b466285e8))
- Updated CI yamls ([edcb92c](https://github.com/southworks/botbuilder-dotnet/pull/458/commits/edcb92c26bcaa2c4489cdcdb86b843ef962a833a))

_Note: The FunctionalTests pipelines will be updated in a separate PR._

## Specific Changes
- **For the test projects:**
   - Update projects to target net6.0 as well as netcoreapp3.1.
   - Fixed [CA1309](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1309) issues with StringComparison.
   - Added a condition to evaluate _lastIndexOf_ in _ExpressionParserTests.cs_ as it's different in Net5 or higher.
   - Adapted test in _ServiceCollectionExtensionsTests.cs_ due to a change in the use of _GetEnumerator_ in _TryAddSingleton_ method.
   - Replaced obsolete _WebClient_ with _HttpClient_ in _WebexClientTests.cs_.

- **For the bots projects:**
   - Updated all the bots to target .NET6 instead of netcoreapp3.1.
   - Fixed [CA1309](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1309) issues with StringComparison.

- **For the CI pipelines yamls:**
   - Updated the _vmImages_ from windows-2019 to windows-2022.
   - Added steps to build and test targeting .NET6.
   - Changed _vsVersion_ from 16.0 to 17.0.
   - Added step to install bf cli in `botbuilder-dotnet-simple-build-test.yml` to avoid the _Microsoft.Bot.Builder.Dialogs.Declarative.Tests_ to fail

## Testing
These images show some of the updated bots working as expected.
image

Here we can see the pipelines working after the changes.
image

### CODE REVIEW

**Comment by sw-joelmut** [comment]
For the commit list i suggest using the commit only like this to reference it, so you dont have to update the url once we promote the pr.

Updated CI yamls (5131f1e72d27d4417c2c8648af7473a48c6760dc)

File: tests/Microsoft.Bot.Builder.AI.Luis.TestUtils/Utils.cs [hunk 1]
```diff
@@ -81,7 +81,7 @@ public static bool WithinDelta(JToken expectedToken, JToken actualToken, double
                 {
                     if (expectedToken.Type == JTokenType.String)
                     {
-                        withinDelta = expectedToken.Value<string>().Equals(actualToken.Value<string>(), StringComparison.InvariantCultureIgnoreCase);
+                        withinDelta = expectedToken.Value<string>().Equals(actualToken.Value<string>(), StringComparison.Ordinal);
```
Comment (sw-joelmut) [inline tests/Microsoft.Bot.Builder.AI.Luis.TestUtils/Utils.cs:84]: This should be OrdinalIgnoreCase as a replacement of InvariantCultureIgnoreCase

File: tests/Microsoft.Bot.Builder.Dialogs.Tests/TextPromptTests.cs [hunk 2]
```diff
@@ -113,8 +113,8 @@ public async Task TextPromptWithNaughtyStrings()
                 {
                     // If the input message is empty after a .Trim() operation, character the comparison will fail because the reply message will be a
                     // Message Activity with null as Text, this is expected behavior
-                    var messageIsBlank = e.Message.Equals(" :\nExpected: \nReceived:", StringComparison.CurrentCultureIgnoreCase) && naughtyString.Equals(" ", StringComparison.CurrentCultureIgnoreCase);
-                    var messageIsEmpty = e.Message.Equals(":\nExpected:\nReceived:", StringComparison.CurrentCultureIgnoreCase) && naughtyString.IsNullOrEmpty();
+                    var messageIsBlank = e.Message.Equals(" :\nExpected: \nReceived:", StringComparison.Ordinal) && naughtyString.Equals(" ", StringComparison.Ordinal);
+                    var messageIsEmpty = e.Message.Equals(":\nExpected:\nReceived:", StringComparison.Ordinal) && naughtyString.IsNullOrEmpty();
```
Comment (sw-joelmut) [inline tests/Microsoft.Bot.Builder.Dialogs.Tests/TextPromptTests.cs:117]: Same as mentioned previously, CurrentCultureIgnoreCase to OrdinalIgnoreCase

File: tests/Microsoft.Bot.Builder.Tests/TranscriptUtilities.cs [hunk 3]
```diff
@@ -62,7 +62,7 @@ public static IEnumerable<IActivity> GetActivities(string relativePath)
             }
 
             string content;
-            if (string.Equals(Path.GetExtension(path), ".chat", StringComparison.InvariantCultureIgnoreCase))
+            if (string.Equals(Path.GetExtension(path), ".chat", StringComparison.Ordinal))
```
Comment (sw-joelmut) [inline tests/Microsoft.Bot.Builder.Tests/TranscriptUtilities.cs:65]: Same here

**Review by sw-joelmut — CHANGES_REQUESTED** [review]
Just minor changes

File: tests/Microsoft.Bot.Builder.AI.Luis.TestUtils/Utils.cs (hunk 1 above)
Comment (ceciliaavila) [reply tests/Microsoft.Bot.Builder.AI.Luis.TestUtils/Utils.cs]: Fixed. Thanks!

File: tests/Microsoft.Bot.Builder.Dialogs.Tests/TextPromptTests.cs (hunk 2 above)
Comment (ceciliaavila) [reply tests/Microsoft.Bot.Builder.Dialogs.Tests/TextPromptTests.cs]: Fixed. Thanks!

File: tests/Microsoft.Bot.Builder.Tests/TranscriptUtilities.cs (hunk 3 above)
Comment (ceciliaavila) [reply tests/Microsoft.Bot.Builder.Tests/TranscriptUtilities.cs]: Fixed. Thanks!

**Review by erquirogasw — APPROVED** [review]

**Review by sw-joelmut — APPROVED** [review]

**Pull request closed without merging**